packages/web-components/src/components/va-button-icon/va-button-icon.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Listen,
+  Prop,
+  Element,
+} from '@stencil/core';
+import classnames from 'classnames';
+
+/**
+ * @componentName Button - Icon
+ * @nativeHandler onClick
+ * @maturityCategory caution
+ * @maturityLevel candidate
+ */
+
+@Component({
+  tag: 'va-button-icon',
+  styleUrl: 'va-button-icon.scss',
+  shadow: true,
+})
+export class VaButtonIcon {
+  @Element() el: HTMLElement;
+
+  /**
+   * If `true`, the component-library-analytics event is disabled.
+   */
+  @Prop() disableAnalytics?: boolean = false;
+
+  /**
+   * If `true`, the click event will not fire.
+   */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /**
+   * The aria-label of the component.
+   */
+  @Prop() label?: string; // could use this.el.getAttribute('aria-label') but this is more explicit
+
+  /**
+   * The type of button this will render as, currently a limited number of options
+   */
+
+  private buttonTypeMap = {
+    // eslint-disable-next-line i18next/no-literal-string
+    'change-file': { icon: 'attach_file', text: 'Change File' },
+    // eslint-disable-next-line i18next/no-literal-string
+    'delete': { icon: 'delete', text: 'Delete' },
+    // eslint-disable-next-line i18next/no-literal-string
+    'cancel': { icon: 'close', text: 'Cancel' },
+  };
+  @Prop() buttonType: keyof typeof this.buttonTypeMap;
+
+  /**
+   * The event used to track usage of the component.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'component-library-analytics',
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleClick = (): void => {
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-button',
+        action: 'click',
+        details: {
+          // eslint-disable-next-line i18next/no-literal-string
+          type: 'primary',
+          label: this.buttonTypeMap[this.buttonType].text,
+        },
+      };
+      this.componentLibraryAnalytics.emit(detail);
```

Comment:
We will want to alert the analytics team that this component has been added. I think they will need to make some updates on their end for it to be tracked. And it will also need to be added to the [vets-website analytics script](https://github.com/department-of-veterans-affairs/vets-website/blob/beb2173be8f06f54050e0f31e61dbcb6bbfbd384/src/platform/site-wide/component-library-analytics-setup.js). Maybe best for a separate ticket?